simulator/source/Combat_simulator.cpp

```diff
@@ -452,6 +454,39 @@ void Combat_simulator::mortal_strike(Sim_state& state)
     logger_.print("Current rage: ", int(rage));
 }
 
+void Combat_simulator::devastate(Sim_state& state)
+{
+    //TODO: Add DPR support
+    logger_.print("Devastate!");
+    int bonus = (config.exposed_armor && apply_delayed_armor_reduction) ? 0 : std::vector<int>{0, 35*2, 35*3, 35*4, 35*5, 35*5}[std::min(sunder_armor_stacks_, 5)];
+    double damage = ((state.main_hand_weapon.normalized_swing(state.special_stats) - state.special_stats.bonus_damage)/2 + bonus) * 1.10; //  devastate not affected by bonus damage
+    const auto& hit_outcome = generate_hit(state, state.main_hand_weapon, hit_table_yellow_mh_, damage);
+    if (hit_outcome.hit_result == Hit_result::miss || hit_outcome.hit_result == Hit_result::dodge)
+    {
+        spend_rage(0.2 * devastate_rage_cost_);
+        if (hit_outcome.hit_result == Hit_result::dodge && has_warbringer_4_set_)
+        {
+            gain_rage(2);
+        }
+    }
+    else
+    {
+        spend_rage(devastate_rage_cost_);
+        maybe_gain_flurry(hit_outcome.hit_result, state.flurry_charges, state.special_stats);
+        hit_effects(state, hit_outcome.hit_result, state.main_hand_weapon);
+        hit_effects(state, hit_outcome.hit_result, state.main_hand_weapon);  //  double proc chance on devastate
```

Comment:
That makes sense, I was only thinking about half the problem. The first issue is whether or not devastate will have 2 proc chances when the target already has 5 stacks of sunder, which Beanna indicated to me that it does (this was in question because devastate doesn't give the threat for that potential sunder cast). The second issue is what you're describing where IEA is applied and sunder is unable to be cast at all, which means both that there will be no extra threat as well as no extra proc chance. I updated the 2nd proc roll to only occur if IEA isn't on the target.